src/_pytest/config/__init__.py

```diff
@@ -351,9 +351,9 @@ def _get_legacy_hook_marks(
     if TYPE_CHECKING:
         # abuse typeguard from importlib to avoid massive method type union thats lacking a alias
         assert inspect.isroutine(method)
-    known_marks: set[str] = {m.name for m in getattr(method, "pytestmark", [])}
-    must_warn: list[str] = []
-    opts: dict[str, bool] = {}
+    known_marks: Set[str] = {m.name for m in getattr(method, "pytestmark", [])}
+    must_warn: List[str] = []
+    opts: Dict[str, bool] = {}
```

Comment:
These changes were requested by mypy on pre-commit hook; could it be that it didn't get the config from the appropriate places?

Reply:
Hmm, the builtin-generics form (lower-case) [requires Python 3.9](https://peps.python.org/pep-0585/), so I think this diff is probably correct but definitely belongs in a separate PR if so 🤔